### PR TITLE
[10.0][ADD] auth_brute_force_oauth

### PR DESCRIPTION
**Pull request by TimLai125**
This PR adds `auth_brute_force_oauth` as a bridge module between `auth_brute_force` and `auth_oauth`.
The module fixes the "Authentication Attempts" records from Oauth login, the records will be showing "Successful" with Oauth login.

Since `auth_brute_force` and `auth_oauth` are independent to each other. The execution order of their `check_credentials` could vary in different environments.

Therefore this module will conbime the `check_credentials` from both modules:
1. First the logic should check against whether the remote is banned.
2. If the user is not banned, check if the uid and password pair match any exist user's Oauth access token.
3. Lastly, the logic should check against user's uid and password.

The `_auth_oauth_signin` needs to be further adjusted as to raise `AccessDenied` during the Oauth login flow. Otherwise, there will not be any error message shown in the screen when the user is banned.

Related Issue: https://github.com/OCA/server-tools/issues/2023

### CODE REVIEW

File: auth_brute_force_oauth/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    'author': "Quartile Limited, "
+              "Odoo Community Association (OCA)",
+    'website': 'https://github.com/OCA/server-tools',
+    'license': 'AGPL-3',
+    'depends': [
+        "auth_oauth",
+        "auth_brute_force",
+    ],
+    'data': [],
+    'installable': True,
```
Comment (legalsylvain) [inline auth_brute_force_oauth/__manifest__.py:17]: if it is fixing an incompatibility between both module, it should be ``auto_install: True``.

**Review by legalsylvain — COMMENTED** [review]
partial review.

**Review by yostashiro — COMMENTED** [review]
Would it be an option to only override `check_credentials()` and keep successful OAuth logins outside of the scope of logging, by inserting
```.py
        res = self.sudo().search([('id', '=', self.env.uid), ('oauth_access_token', '=', password)])
        if not res:
```
at the beginning?

I personally think logging of successful attempts is not important for OAuth logins (much less vulnerable to brute-force attacks than password logins) and we should still be able to capture failed attempts if I'm not mistaken.

**Review by legalsylvain — COMMENTED** [review]
sorry I don't use oauth, so I can not review this PR.

thanks for this contribution anyway.

File: auth_brute_force_oauth/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Authentification - Brute-Force Filter - (Oauth Support)',
```
Comment (yostashiro) [inline auth_brute_force_oauth/__manifest__.py]:
```suggestion
    'name': 'Authentification - Brute-Force Filter (Oauth Support)',
```

File: auth_brute_force_oauth/models/res_users_hook_check_credentials.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        )
+        if not trusted:
+            error = AccessDenied()
+            error.reason = "banned"
+            raise error
+        res = self.sudo().search([
+            ('id', '=', self.env.uid),
+            ('oauth_access_token', '=', password)
+        ])
+        if res:
```
Comment (yostashiro) [inline auth_brute_force_oauth/models/res_users_hook_check_credentials.py]:
```suggestion
        if self.sudo().search([
            ('id', '=', self.env.uid),
            ('oauth_access_token', '=', password)
        ]):
```

**Review by yostashiro — APPROVED** [review]
Functional review. LGTM. 👍  Just some minor non-blocking comments.

**Pull request closed without merging**